shared/api/storage.go

```diff
@@ -0,0 +1,37 @@
+package api
+
+// StoragePoolConfig represents the fields of a LXD storage pool.
+type StoragePool struct {
+	PoolName       string   `json:"name" yaml:"name"`
+	PoolDriver     string   `json:"driver" yaml:"driver"`
+	PoolUsedBy     []string `json:"used_by" yaml:"used_by"`
+	StoragePoolPut `yaml:",inline"`
+}
+
+type StoragePoolPut struct {
+	PoolConfig map[string]string `json:"config" yaml:"config"`
+}
+
+// StorageVolumeConfig represents the fields of a LXD storage volume.
+type StorageVolume struct {
+	VolumeType       string   `json:"type" yaml:"type"`
+	VolumeUsedBy     []string `json:"used_by" yaml:"used_by"`
+	StorageVolumePut `yaml:",inline"`
+}
+
+type StorageVolumePut struct {
+	VolumeName   string            `json:"name" yaml:"name"`
+	VolumeConfig map[string]string `json:"config" yaml:"config"`
+}
+
+// Writable converts a full StoragePool struct into a StoragePoolPut struct
+// (filters read-only fields).
+func (storagePool *StoragePool) Writable() StoragePoolPut {
+	return storagePool.StoragePoolPut
+}
+
+// Writable converts a full StorageVolume struct into a StorageVolumePut struct
+// (filters read-only fields).
+func (storageVolume *StorageVolume) Writable() StorageVolumePut {
+	return storageVolume.StorageVolumePut
+}
```

Comment:
Please ensure that shared/api/ remains "golint" clean as that's the only way to ensure that the godoc output for the API remains valid.